File: goibackend/src/main/java/com/sebastianaldi17/goibackend/models/Vocab.java
package com.sebastianaldi17.goibackend.models;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.persistence.*;

import lombok.Data;

import java.util.List;

@Entity(name = "vocabs")
@Data
public class Vocab {
    @Id
    @GeneratedValue
    @JsonIgnore
    private Long id;

    private String kanji;
    private String kana;
    private String level;

    @OneToMany(mappedBy = "vocab")
    @JsonManagedReference
    private List<Example> examples;

    @OneToMany(mappedBy = "vocab")
    @JsonManagedReference
    private List<Definition> definitions;
}

File: goiweb/classes/vocab.ts
export type Vocab = {
    definitions: Definition[];
    examples: Example[];
    kana: string;
    kanji: string;
    level: string;
    id: number;
}

export type Definition = {
    meanings: string[];
    parts_of_speech: string[];
    tags: string[];
}

export type Example = {
    english: string;
    japanese: string;
}